### PR TITLE
Add test cases for loading dynamic models and convert.jl

### LINKED ISSUE #56: Find and add good tests for loading converters
> is there some good way/data to test this?

I created these test for the converters: https://github.com/paulflang/SBML.jl/blob/pl/mk-species-units/test/loaddynamicmodels.jl

_Originally posted by @paulflang in https://github.com/LCSB-BioCore/SBML.jl/issues/55#issuecomment-855837778_

### PR DESCRIPTION
**Pull request by giordano**
Revive #64, possibly fix #56.  CC: @paulflang

### CODE REVIEW

File: test/loaddynamicmodels.jl [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        ),
+        (
+            joinpath(@__DIR__, "data", "reactionsystem_03.xml"),
+            ["expandInitialAssignments" => Dict{String,String}()],
+            1,
+            ("reaction_2", c1*k1*s4)
+        ),
+        (
+            joinpath(@__DIR__, "data", "reactionsystem_04.xml"),
+            # PL: TODO: Unfortunately, libSBML has an inconsistent API here I think. Hence
```
Comment (exaexa) [inline test/loaddynamicmodels.jl:35]: could we get a hint about what's "broken" if there's no level set?

**Comment by giordano** [comment]
Ok, I had started this PR long time ago and it looks like I had also figured out where the input files were.  Tests are passing locally for me, but now I need to remember where I downloaded them from :grimacing:

**Comment by giordano** [comment]
For my own record: the content of #64, before a force-push made its content completely disappear, was https://github.com/LCSB-BioCore/SBML.jl/commit/31acf728eeca28bf11b8ca9098e01f6426c8d088, which has somewhat different tests.

The tests I used here are from https://github.com/paulflang/SBML.jl/blob/pl/mk-species-units/test/loaddynamicmodels.jl, mentioned in #56.

File: test/loaddynamicmodels.jl [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            @test length(mdl.parameters) == expected_par
+
+            id, math = expected_rxn
+            @test isequal(repr(mdl.reactions[id].kinetic_math), repr(math))
+
+            id, ic = expected_u0
+            if basename(sbmlfile) == "00037-sbml-l3v2.xml"
+                # When expanding initial assignments with libsbml, the initial amount
+                # becomes empty.
+                @test_broken mdl.species[id].initial_amount[1] == ic
```
Comment (giordano) [inline test/loaddynamicmodels.jl:85]: This is basically what we found in #163

**Review by exaexa — APPROVED** [review]
looks good, let's see. :D

**Pull request merged**